karyon3-admin-simple/src/main/java/com/netflix/karyon/admin/rest/AdminHttpHandler.java

```diff
@@ -99,10 +101,17 @@ private boolean shouldUseGzip(HttpExchange exchange) {
         return (encoding != null && encoding.toLowerCase().contains("gzip"));
     }
 
-    private OutputStream getOutputStream(boolean useGzip, HttpExchange exchange) throws IOException {
-        return useGzip
-            ? new GZIPOutputStream(exchange.getResponseBody())
-            : exchange.getResponseBody();
+    private byte[] compress(byte[] input) throws IOException {
```

Comment:
Not sure that I like the idea of making an extra copy.  We should revisit this at some point.